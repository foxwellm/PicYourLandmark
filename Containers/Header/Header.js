import React, { Component } from 'react';
import { StyleSheet, Text, View} from 'react-native';
import { Icon } from 'react-native-elements';

export class Header extends Component {
  constructor() {
    super()
    this.state = {
      titleText: 'PIC YOUR LANDMARK'
    }
  }

  render() {
    const {titleText} = this.state
    return (
      <View style={styles.container}>
        <Icon name="map-marker" type="font-awesome" size={22} color='tomato' position='absolute' top={-17} left={'-25%'}/>
        <Text style={styles.titleText}>
          {titleText}
        </Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    backgroundColor: '#e9e9e9',
    flexDirection: 'column',
    alignItems: 'center',
    flex: .4,
    justifyContent: 'flex-end'

  },
  titleText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#3c4859',
  }
});

export default Header;