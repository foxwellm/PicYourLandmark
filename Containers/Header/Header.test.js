import React from 'react';
import { Header } from './Header';
import renderer from 'react-test-renderer';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
