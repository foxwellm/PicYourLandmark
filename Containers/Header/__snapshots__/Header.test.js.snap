// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e9e9e9",
      "display": "flex",
      "flex": 0.4,
      "flexDirection": "column",
      "justifyContent": "flex-end",
    }
  }
>
  <View>
    <View
      left="-25%"
      position="absolute"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
      top={-17}
      underlayColor="white"
    >
      <Text />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#3c4859",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    PIC YOUR LANDMARK
  </Text>
</View>
`;

exports[`Header should match the snapshot with the correct title text 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e9e9e9",
      "display": "flex",
      "flex": 0.4,
      "flexDirection": "column",
      "justifyContent": "flex-end",
    }
  }
>
  <Themed.Icon
    color="tomato"
    left="-25%"
    name="map-marker"
    position="absolute"
    size={22}
    top={-17}
    type="font-awesome"
  />
  <Component
    style={
      Object {
        "color": "#3c4859",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    PIC YOUR LANDMARK
  </Component>
</Component>
`;
